framework/auth/views.py

```diff
@@ -190,6 +192,27 @@ def auth_logout(redirect_url=None):
     return resp
 
 
+def auth_email_logout(token, user):
+    """When a user is adding an email or merging an account, add the email to the user and log them out.
+    """
+    redirect_url = web_url_for('auth_login') + '?existing_user={}'.format(user.email)
+    if user.confirm_token(token):
+        try:
+            user_merge = User.find_one(Q('emails', 'eq', user.email_verifications[token]['email'].lower()))
```

Comment:
Since I think this action is rare and the tokens are pretty long and expire quickly, I wouldn't think it would be a problem.